From dd0f8efbb4b96054a849fee0cb57faeaa421a537 Mon Sep 17 00:00:00 2001
From: JC Fillion-Robin <jcfr@nvidia.com>
Date: Tue, 10 Mar 2026 15:43:54 -0400
Subject: [PATCH 04/10] Build python libs with install RPATH and add find

Patch-name: 0004-Build-python-libs-with-install-RPATH-and-add-find-py.patch
Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Ian Stewart <istewart@nvidia.com>
Signed-off-by: Ilies CHERGUI <ichergui@nvidia.com>
---
 cmake/pybind11_add_holohub_module.cmake | 15 +++------------
 1 file changed, 3 insertions(+), 12 deletions(-)

diff --git a/cmake/pybind11_add_holohub_module.cmake b/cmake/pybind11_add_holohub_module.cmake
index 73ec6bd1..c1b73ef7 100644
--- a/cmake/pybind11_add_holohub_module.cmake
+++ b/cmake/pybind11_add_holohub_module.cmake
@@ -16,15 +16,7 @@
 # Find pybind11
 find_package(Python3 REQUIRED COMPONENTS Interpreter Development.Module)
 
-# We fetch pybind11 since we need the same version as the Holoscan SDK
-# and it's not necessarily available on all the platforms
-include(FetchContent)
-FetchContent_Declare(pybind11
-  GIT_REPOSITORY https://github.com/pybind/pybind11
-  GIT_TAG v2.13.6
-  GIT_SHALLOW TRUE
-)
-FetchContent_MakeAvailable(pybind11)
+find_package(pybind11 2.11.1 REQUIRED)
 
 # Helper function to generate pybind11 operator modules
 function(pybind11_add_holohub_module)
@@ -123,9 +115,8 @@ function(pybind11_add_holohub_module)
         "\$ORIGIN/../lib"    # legacy fallback for one-level-deep layouts
     )
     list(JOIN _rpath ":" _rpath)
-    set_property(TARGET ${target_name}
-        APPEND PROPERTY BUILD_RPATH ${_rpath}
-    )
+    set_target_properties(${target_name} PROPERTIES INSTALL_RPATH ${_rpath})
+    set_target_properties(${target_name} PROPERTIES BUILD_WITH_INSTALL_RPATH ON)
     unset(_rpath)
 
     # make submodule folder
